@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -575,9 +656,11 @@ function placePointLabelAtCurrentAnchor(
         textCanvas.measureText(label.glyphs!, label.bounds, tmpMeasurementParams);
     }
 
+    // Revert back text canvas layout.
+    textCanvas.textLayoutStyle.horizontalAlignment = baseAlign.h;
+    textCanvas.textLayoutStyle.verticalAlignment = baseAlign.v;
```

Comment:
no need to restore the base alignment. If the label is successfully placed, it will be with the last alignment set on the text canvas, if it's rejected, the last aligment set won't be used at all.

Reply:
Wrong! It will be used... the last successful placement will be used even if label is rejected. I have a strong feeling that this is a basic problem of our misunderstanding:
**Label that are rejected will be rendered few more frames with FADE OUT effect and require to KEEP their layout consistent - no more multi-anchor placement **

Reply:
But here you're reverting back to the base aligment, instead of the last aligment used. If placement fails and the label is rejected it will start fading out in the base aligment instead of using the last aligment?
I think we need to discuss this online, I'm really confused.

Reply:
Yep I see, that after refactoring I have created a bug here -> it should be `lastPlacement` instead of base one.
Anyway I have change the logic as you suggested, so TextCanvas layout state is reverted only on demand in `placePointLabelChoosingAnchor()`, this looks simpler. If I will need to optimize later as discussed previously I will figure out something better.